internal/pkg/xds/adsc/adsc.go

```diff
@@ -80,6 +80,14 @@ func (a *ADSC) Run() error {
 	return nil
 }
 
+func (a *ADSC) Restart() {
+	log.Infof("reconnecting to ADS server %s", a.cfg.DiscoveryAddr)
+	if err := a.Run(); err != nil {
+		log.Errorf("failed to start ADS server %s, will reconnect in %s: %v", a.cfg.DiscoveryAddr, a.cfg.ReconnectDelay, err)
```

Comment:
Or "failed to start ADS client"?
```suggestion
		log.Errorf("failed to connect to ADS server %s, will reconnect in %s: %v", a.cfg.DiscoveryAddr, a.cfg.ReconnectDelay, err)
```